app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/ConfirmDialog.java

```diff
@@ -0,0 +1,67 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatDialogFragment;
+
+import com.cmput301f20t21.bookfriends.R;
+
+import java.util.List;
+
+public class ConfirmDialog extends AppCompatDialogFragment {
+    private ConfirmDialogListener confirmDialogListener;
+    private String requesterId;
+    private List<String> idsToRemove;
+
+    public ConfirmDialog(String id, List<String> idsToRemove) {
+        this.requesterId = id;
+        this.idsToRemove = idsToRemove;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        LayoutInflater inflater = getActivity().getLayoutInflater();
+        View v = inflater.inflate(R.layout.confirm_dialog, null);
+        builder.setView(v)
+                .setTitle(getString(R.string.accept_this_request))
+                .setNegativeButton(getString(R.string.edit_cancel), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        // does not do anything
+                    }
+                })
+                .setPositiveButton(getString(R.string.edit_confirm), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        confirmDialogListener.setConfirm(requesterId, idsToRemove);
+                    }
+                });
+        return builder.create();
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+
+        try {
+            confirmDialogListener = (ConfirmDialogListener) context;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(context.toString() +
+                    "Must implement ConfirmDialogListener");
+        }
+    }
+
+    public interface ConfirmDialogListener {
+        void setConfirm(String id, List<String> idsToRemove);
```

Comment:
rename to onConfirm

Reply:
On this, I think we should avoid implementing a listener only for that single confirm callback. Instead, when creating the confirm fragment, pass an onConfirm callback (preferably a lambda) to the fragment. We don't even have to use a new interface for the callback, could use `OnSuccessCallbackWithMessage` with a type parameter `<List<String>>` as the idsToRemove list.

Reply:
sorry, now looking back on this, in my opinion we don't need any parameters at all, do we? the id is contained in view model when passing the callback and the idsToRemove is calculated by the view model.

Reply:
I don't think there's a clean way to pass onConfirm callback to fragment as param? Declaring a new interface is fine here too since OnSuccessCallback doesn't really fit here. Nothing has success yet but confirming (then we begin to make call to the db)

Reply:
unfortunately I dived too deep on my refactor commit and made it working without this class. are you looking at the latest code?